mlflow/pyfunc/__init__.py

```diff
@@ -662,31 +701,23 @@ def predict(self, data: PyFuncInput, params: Optional[Dict[str, Any]] = None) ->
         Returns:
             Model predictions as one of pandas.DataFrame, pandas.Series, numpy.ndarray or list.
         """
-        input_schema = self.metadata.get_input_schema()
-        flavor = self.loader_module
-        if input_schema is not None:
-            try:
-                data = _enforce_schema(data, input_schema, flavor)
-            except Exception as e:
-                # Include error in message for backwards compatibility
-                raise MlflowException.invalid_parameter_value(
-                    f"Failed to enforce schema of data '{data}' "
-                    f"with schema '{input_schema}'. "
-                    f"Error: {e}",
-                )
-
-        params = _validate_params(params, self.metadata)
 
+        data, params = self._validate_prediction_input(data, params)
         if inspect.signature(self._predict_fn).parameters.get("params"):
             return self._predict_fn(data, params=params)
-        _log_warning_if_params_not_in_predict_signature(_logger, params)
-        if HAS_PYSPARK and isinstance(data, SparkDataFrame):
-            _logger.warning(
-                "Input data is a Spark DataFrame. Note that behaviour for "
-                "Spark DataFrames is model dependent."
-            )
         return self._predict_fn(data)
 
+    def predict_stream(
+        self, data: PyFuncInput, params: Optional[Dict[str, Any]] = None
+    ) -> Iterator[PyFuncOutput]:
```

Comment:
Are there plans to eventually expose `convert_chat_responses` here? seems like it would make sense for predict_stream use cases

Reply:
We can add `convert_chat_responses` as an inference param.

Note that `model.predict` can also have `convert_chat_responses` param , I can do it in the follow-up PR, to add them together

Reply:
so we add `convert_xxx_responses` every time we add a new response format?

Reply:
> so we add a convert_xxx_responses argument every time we add a new response format?

We can discuss it in follow-up PR. Maybe we need a better design.

Reply:
sgtm, we'll can use the internal `_predict_stream_with_callbacks` for now. @BenWilson2 have you heard of any customers request OpenAI compatible responses for mlflow.predict? Model serving is thinking OpenAI API format in -> OpenAI API format out would make things simple to think about on the serving side and align with existing products more